Add captioning support for video/audio files

yingjin opened the pull request:
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes https://github.com/DSpace/dspace-angular/issues/1969

## Description
Add captioning support for video/audio files. Only front end angular is changed. 

## Instructions for Reviewers
Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

List of changes in this PR:
* Add language helper contains a const object mapping of two letter with their Full language name
* Add a capInfos class to save related info to captioning
* Updated several .ts and .html files

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes. 

Have an item with media bitstreams, either video/audio files, and their corresponding .vtt file named following this pattern -

the video file "greenteam.mp4" media should have a .vtt file named "greenteam.mp4-en.vtt". -en is the two letter language code and will be translated to full language name using language helper. You can have multiple .vtt files associated with it if there are multiple languages. 

Refer language code here or language-helper.ts file.
https://www.w3schools.com/tags/ref_language_codes.asp

You also need to turn on the media viewer in config.yml, add new format at registries -> format. Name: WEB VTT, minetype: text/vtt and suffix: vtt.

After this, you could ingest the items with media items and vtt files and test it in media viewer. It should show you the caption off when click three dot at the right bottom corner of the media player. Choose caption and you should see a list of the language(s) listed. Choose one and test it.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [ x ] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [ ] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

jtimal commented:
Hi @yingjin 

I checked the operation and it works very well and I can change languages and everything is excellent.
![image](https://user-images.githubusercontent.com/30701057/203428694-0fd99b83-8da1-4b0d-8d74-ec8f604284c7.png)

tdonohue commented on src/app/item-page/media-viewer/media-viewer-video/caption-info.ts:
```diff
@@ -0,0 +1,4 @@
+export class CaptionInfo {
```
Could we add some TypeDocs / comments to better describe this class?  It looks like its purpose is to store information about the captioning file, so that it can be placed in the HTML5 `<track>` tag.  So, we should add a brief note to describe how this is used.

tdonohue commented on src/app/item-page/media-viewer/media-viewer-video/media-viewer-video.component.html:
```diff
@@ -8,7 +9,15 @@
   controls
-></video>
+>
+<!--  <track default kind="captions" src="./assets/vtt/JusticeAndTransformation_caption.mp4-en.vtt" label="English" type="text/vtt"/>-->
```
Small thing, could we remove this commented out code?

tdonohue commented on src/app/item-page/media-viewer/media-viewer-video/media-viewer-video.component.ts:
```diff
@@ -26,9 +28,39 @@ export class MediaViewerVideoComponent implements OnInit {
+    let filteredCapMedias: MediaViewerItem[];
+    let capInfos: CaptionInfo[] = [];
+    filteredCapMedias = this.medias
+       .filter((media) => media.minetype === 'text/vtt')
```
We may want to add the 'text/vtt' mime type to the DSpace registry by default (this would require a separate backend PR).  That way it's easier for others to reuse this code.  For example, if someone forgets to add this mimeType & then uploads a VTT file, captioning won't work.

tdonohue commented on src/app/item-page/media-viewer/media-viewer-video/media-viewer-video.component.ts:
```diff
@@ -26,9 +28,39 @@ export class MediaViewerVideoComponent implements OnInit {
+      filteredCapMedias
+        .forEach((media, index) => {
+          let srclang: string = media.bitstream.name.slice(-6, -4).toLowerCase();
+          //capInfos[index] = new CaptionInfo(
```
Again, it'd be best to remove any commented out code.

tdonohue reviewed (approved):
@yingjin : Thanks for this improvement!  I gave this a test and was able to successfully get captioning working on a video using a `*.vtt` file.  

The only thing that tripped me up was the naming convention...I had a video named `test.mp4` and I originally named the VTT `test-en.vtt` only to realize it _should be named_ `test.mp4-en.vtt`.  That's just my own mistake, but we may want to make sure to write some clear documentation on this naming convention.

All that said, I have a few minor comments below on things to clean up in the code.   I also think we should create a small PR for the backend to add VTT to our DSpace Format Registry -- otherwise, I'd worry someone will forget that step and find captioning doesn't work.

atarix83 reviewed (changes requested):
thanks @yingjin for this PR

the code looks good, but i can't find a way to test it. I've followed instructions but the captions didn't appear. 
I've created an item in the demo here https://demo7.dspace.org/items/0220cb00-3eb9-467d-916c-2ab588caa9b7

could you tell me what is wrong ?

moreover while i was trying to enabled the media viewer i found a bug that is not related to this PR, but it'd be nice if you can fix.
Basically if you enable the media viewer only for video the media player is not shown
```
mediaViewer:
  image: false
  video: true
```
It's enough to fix the following lines 
https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/publication/publication.component.html#L27
https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/untyped-item/untyped-item.component.html#L27

yingjin commented:
Thanks for testing it, Julian!

On Tue, Nov 22, 2022 at 3:53 PM Julian Timal ***@***.***>
wrote:

> Hi @yingjin
> <https://urldefense.com/v3/__https://github.com/yingjin__;!!BuQPrrmRaQ!gcdreVSt7iXBhKGuejN6u0--C6qtE-ZnjAhgpx1XwHSSy95Zjzl8WCCntD2nw5vjYLpsM5_gtOpEbE2qX5rCNAGUkEbEJg$>
>
> I checked the operation and it works very well and I can change languages
> and everything is excellent.
> [image: image]
> <https://urldefense.com/v3/__https://user-images.githubusercontent.com/30701057/203428694-0fd99b83-8da1-4b0d-8d74-ec8f604284c7.png__;!!BuQPrrmRaQ!gcdreVSt7iXBhKGuejN6u0--C6qtE-ZnjAhgpx1XwHSSy95Zjzl8WCCntD2nw5vjYLpsM5_gtOpEbE2qX5rCNAGfJ4dOcA$>
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/DSpace/dspace-angular/pull/1967*issuecomment-1324281794__;Iw!!BuQPrrmRaQ!gcdreVSt7iXBhKGuejN6u0--C6qtE-ZnjAhgpx1XwHSSy95Zjzl8WCCntD2nw5vjYLpsM5_gtOpEbE2qX5rCNAHNt_-eGw$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AALAW7T647XKWFDFQIFWNXDWJU6FBANCNFSM6AAAAAASCS3YH4__;!!BuQPrrmRaQ!gcdreVSt7iXBhKGuejN6u0--C6qtE-ZnjAhgpx1XwHSSy95Zjzl8WCCntD2nw5vjYLpsM5_gtOpEbE2qX5rCNAER4VuNxg$>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

yingjin commented:
Hi Tim,

Thank you for the suggestions. I added comments for CapInfo.ts class and
removed the comments in the other two files.

It would be nice to add text/vtt mime type to the registry by default.
Also, jpeg 2000 wasn't in the registry for IIIF viewer. If it is still the
case, might add that as well.

Best,
Ying

On Wed, Dec 14, 2022 at 4:45 PM Tim Donohue ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> @yingjin
> <https://urldefense.com/v3/__https://github.com/yingjin__;!!BuQPrrmRaQ!haT-xVz3GzN4uOBdUoa1V7fcn3IkSrcLjKihaoR0c9jc7Wr2MBXlxCDv4sGdVOb3exBzBOK3qtDJnlMNeAgx3-1-SfOahg$>
> : Thanks for this improvement! I gave this a test and was able to
> successfully get captioning working on a video using a *.vtt file.
>
> The only thing that tripped me up was the naming convention...I had a
> video named test.mp4 and I originally named the VTT test-en.vtt only to
> realize it *should be named* test.mp4-en.vtt. That's just my own mistake,
> but we may want to make sure to write some clear documentation on this
> naming convention.
>
> All that said, I have a few minor comments below on things to clean up in
> the code. I also think we should create a small PR for the backend to add
> VTT to our DSpace Format Registry -- otherwise, I'd worry someone will
> forget that step and find captioning doesn't work.
> ------------------------------
>
> In src/app/item-page/media-viewer/media-viewer-video/caption-info.ts
> <https://urldefense.com/v3/__https://github.com/DSpace/dspace-angular/pull/1967*discussion_r1049026407__;Iw!!BuQPrrmRaQ!haT-xVz3GzN4uOBdUoa1V7fcn3IkSrcLjKihaoR0c9jc7Wr2MBXlxCDv4sGdVOb3exBzBOK3qtDJnlMNeAgx3-0p8N7drA$>
> :
>
> > @@ -0,0 +1,4 @@
> +export class CaptionInfo {
>
> Could we add some TypeDocs / comments to better describe this class? It
> looks like its purpose is to store information about the captioning file,
> so that it can be placed in the HTML5 <track> tag. So, we should add a
> brief note to describe how this is used.
> ------------------------------
>
> In
> src/app/item-page/media-viewer/media-viewer-video/media-viewer-video.component.html
> <https://urldefense.com/v3/__https://github.com/DSpace/dspace-angular/pull/1967*discussion_r1049029099__;Iw!!BuQPrrmRaQ!haT-xVz3GzN4uOBdUoa1V7fcn3IkSrcLjKihaoR0c9jc7Wr2MBXlxCDv4sGdVOb3exBzBOK3qtDJnlMNeAgx3-37QypjHA$>
> :
>
> > @@ -8,7 +9,15 @@
>    "
>    preload="none"
>    controls
> -></video>
> +>
> +<!--  <track default kind="captions" src="./assets/vtt/JusticeAndTransformation_caption.mp4-en.vtt" label="English" type="text/vtt"/>-->
>
> Small thing, could we remove this commented out code?
> ------------------------------
>
> In
> src/app/item-page/media-viewer/media-viewer-video/media-viewer-video.component.ts
> <https://urldefense.com/v3/__https://github.com/DSpace/dspace-angular/pull/1967*discussion_r1049030883__;Iw!!BuQPrrmRaQ!haT-xVz3GzN4uOBdUoa1V7fcn3IkSrcLjKihaoR0c9jc7Wr2MBXlxCDv4sGdVOb3exBzBOK3qtDJnlMNeAgx3-0L6ECc-g$>
> :
>
> > +
> +  /**
> +   * This method check if there is caption file for the media
> +   * The caption file name is the media name plus "-" following two letter
> +   * language code and .vtt suffix
> +   *
> +   * html5 video only support WEBVTT format
> +   *
> +   * Two letter language code reference
> +   * https://www.w3schools.com/tags/ref_language_codes.asp <https://urldefense.com/v3/__https://www.w3schools.com/tags/ref_language_codes.asp__;!!BuQPrrmRaQ!haT-xVz3GzN4uOBdUoa1V7fcn3IkSrcLjKihaoR0c9jc7Wr2MBXlxCDv4sGdVOb3exBzBOK3qtDJnlMNeAgx3-0u0CdbSg$>
> +   */
> +  getMediaCap(name: string): CaptionInfo[] {
> +    let filteredCapMedias: MediaViewerItem[];
> +    let capInfos: CaptionInfo[] = [];
> +    filteredCapMedias = this.medias
> +       .filter((media) => media.minetype === 'text/vtt')
>
> We may want to add the 'text/vtt' mime type to the DSpace registry by
> default (this would require a separate backend PR). That way it's easier
> for others to reuse this code. For example, if someone forgets to add this
> mimeType & then uploads a VTT file, captioning won't work.
> ------------------------------
>
> In
> src/app/item-page/media-viewer/media-viewer-video/media-viewer-video.component.ts
> <https://urldefense.com/v3/__https://github.com/DSpace/dspace-angular/pull/1967*discussion_r1049042287__;Iw!!BuQPrrmRaQ!haT-xVz3GzN4uOBdUoa1V7fcn3IkSrcLjKihaoR0c9jc7Wr2MBXlxCDv4sGdVOb3exBzBOK3qtDJnlMNeAgx3-34ULcnOQ$>
> :
>
> > +   *
> +   * Two letter language code reference
> +   * https://www.w3schools.com/tags/ref_language_codes.asp <https://urldefense.com/v3/__https://www.w3schools.com/tags/ref_language_codes.asp__;!!BuQPrrmRaQ!haT-xVz3GzN4uOBdUoa1V7fcn3IkSrcLjKihaoR0c9jc7Wr2MBXlxCDv4sGdVOb3exBzBOK3qtDJnlMNeAgx3-0u0CdbSg$>
> +   */
> +  getMediaCap(name: string): CaptionInfo[] {
> +    let filteredCapMedias: MediaViewerItem[];
> +    let capInfos: CaptionInfo[] = [];
> +    filteredCapMedias = this.medias
> +       .filter((media) => media.minetype === 'text/vtt')
> +       .filter((media) => media.bitstream.name.substring(0, (media.bitstream.name.length - 7) ).toLowerCase() === name.toLowerCase());
> +
> +    if (filteredCapMedias) {
> +      filteredCapMedias
> +        .forEach((media, index) => {
> +          let srclang: string = media.bitstream.name.slice(-6, -4).toLowerCase();
> +          //capInfos[index] = new CaptionInfo(
>
> Again, it'd be best to remove any commented out code.
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/DSpace/dspace-angular/pull/1967*pullrequestreview-1218329344__;Iw!!BuQPrrmRaQ!haT-xVz3GzN4uOBdUoa1V7fcn3IkSrcLjKihaoR0c9jc7Wr2MBXlxCDv4sGdVOb3exBzBOK3qtDJnlMNeAgx3-3qoG9uwg$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AALAW7UAC6J3W25XHUUZKP3WNJEWHANCNFSM6AAAAAASCS3YH4__;!!BuQPrrmRaQ!haT-xVz3GzN4uOBdUoa1V7fcn3IkSrcLjKihaoR0c9jc7Wr2MBXlxCDv4sGdVOb3exBzBOK3qtDJnlMNeAgx3-0hmV1CbA$>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

yingjin commented:
Hi ***@***.***,*

*I am not sure if the feature is in the demo site yet. You can test it
using my branch DA-8586.*

*Thanks for pointing out the image viewer configuration issue.*

*Tim, is that okay for me to fix that in this PR?*

*Best,*
*Ying*

On Thu, Dec 15, 2022 at 10:39 AM Giuseppe ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> thanks @yingjin
> <https://urldefense.com/v3/__https://github.com/yingjin__;!!BuQPrrmRaQ!lp5S3fPx8rLcpJXACMFTf-mptrKGdZLANvC_pEb8cDy7IrXiah_-zkorEs20XjjGxoS9K4dEivtFXEpQjS2CCekMndaSmQ$>
> for this PR
>
> the code looks good, but i can't find a way to test it. I've followed
> instructions but the captions didn't appear.
> I've created an item in the demo here
> https://demo7.dspace.org/items/0220cb00-3eb9-467d-916c-2ab588caa9b7
> <https://urldefense.com/v3/__https://demo7.dspace.org/items/0220cb00-3eb9-467d-916c-2ab588caa9b7__;!!BuQPrrmRaQ!lp5S3fPx8rLcpJXACMFTf-mptrKGdZLANvC_pEb8cDy7IrXiah_-zkorEs20XjjGxoS9K4dEivtFXEpQjS2CCenpoNyZeA$>
>
> could you tell me what is wrong ?
>
> moreover while i was trying to enabled the media viewer i found a bug that
> is not related to this PR, but it'd be nice if you can fix.
> Basically if you enable the media viewer only for video the media player
> is not shown
>
> mediaViewer:
>   image: false
>   video: true
>
> It's enough to fix the following lines
>
> https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/publication/publication.component.html#L27
> <https://urldefense.com/v3/__https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/publication/publication.component.html*L27__;Iw!!BuQPrrmRaQ!lp5S3fPx8rLcpJXACMFTf-mptrKGdZLANvC_pEb8cDy7IrXiah_-zkorEs20XjjGxoS9K4dEivtFXEpQjS2CCelWL6TSXw$>
>
> https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/untyped-item/untyped-item.component.html#L27
> <https://urldefense.com/v3/__https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/untyped-item/untyped-item.component.html*L27__;Iw!!BuQPrrmRaQ!lp5S3fPx8rLcpJXACMFTf-mptrKGdZLANvC_pEb8cDy7IrXiah_-zkorEs20XjjGxoS9K4dEivtFXEpQjS2CCemWteWzXg$>
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/DSpace/dspace-angular/pull/1967*pullrequestreview-1219592488__;Iw!!BuQPrrmRaQ!lp5S3fPx8rLcpJXACMFTf-mptrKGdZLANvC_pEb8cDy7IrXiah_-zkorEs20XjjGxoS9K4dEivtFXEpQjS2CCem4B8K_Vw$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AALAW7WLWIRQOV25MW3KHY3WNNCUJANCNFSM6AAAAAASCS3YH4__;!!BuQPrrmRaQ!lp5S3fPx8rLcpJXACMFTf-mptrKGdZLANvC_pEb8cDy7IrXiah_-zkorEs20XjjGxoS9K4dEivtFXEpQjS2CCenBfFoNLA$>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

tdonohue commented:
@yingjin : If you wanted to fix the image viewer issue in this PR, it's OK with me.  It's not *required* for this PR, but you can do it if you want.

tdonohue commented:
@atarix83 :  I retested this PR with your example on the demo site & it works fine for me.  I suspect maybe you forgot to turn on the captioning in the video viewer?  By default, captioning is always off... but if you click the three-dot icon (lower left) you'll see the "Captions" menu (see screenshot).  Click it and captioning will be enabled.

![screenshot](https://user-images.githubusercontent.com/483997/208150923-fce7b041-1160-4957-bd90-bab76ba3ad17.png)

So, with this PR, enabling captioning involves two main steps...
1) First, obviously you have to have the video viewer enabled (and the image viewer cause of that bug you found):
```
mediaViewer:
  image: true
  video: true
```
2) Then, in the video viewer, you have to turn on captioning by clicking that "Captions" menu.

yingjin commented:
As @atarix83 suggested, 
[https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/publication/publication.component.html#L27](https://urldefense.com/v3/__https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/publication/publication.component.html*L27__;Iw!!BuQPrrmRaQ!lp5S3fPx8rLcpJXACMFTf-mptrKGdZLANvC_pEb8cDy7IrXiah_-zkorEs20XjjGxoS9K4dEivtFXEpQjS2CCelWL6TSXw$)
[https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/untyped-item/untyped-item.component.html#L27](https://urldefense.com/v3/__https://github.com/yingjin/dspace-angular/blob/DA-8586/src/app/item-page/simple/item-types/untyped-item/untyped-item.component.html*L27__;Iw!!BuQPrrmRaQ!lp5S3fPx8rLcpJXACMFTf-mptrKGdZLANvC_pEb8cDy7IrXiah_-zkorEs20XjjGxoS9K4dEivtFXEpQjS2CCemWteWzXg$)

And setup 
mediaViewer:
  image: false
  video: true

Here are the result after changing the line 27, 
from 
<ng-container *ngIf="mediaViewer.image">  
to 
<ng-container *ngIf="mediaViewer.video"> 

<img width="677" alt="Screen Shot 2022-12-16 at 5 59 51 PM" src="https://user-images.githubusercontent.com/1444734/208221351-f02ec0be-c1bd-48dd-8b3f-c5d8475095f9.png">

<img width="1107" alt="Screen Shot 2022-12-16 at 5 58 43 PM" src="https://user-images.githubusercontent.com/1444734/208221358-ff83f0fc-e6fc-4630-90a4-7066ae8c0bc6.png">


Is that expected behavior? 

Ying

tdonohue commented on src/app/core/shared/media-viewer-item.model.ts:
```diff
@@ -14,6 +14,11 @@ export class MediaViewerItem {
+  /**
+   * Incoming Bitsream format mime type
+   */
+  minetype: string;
```
@yingjin : There's a spelling error here.  You've accidentally spelled it `minetype` instead of `mimetype`.  Could you quickly correct this?

yingjin commented:
Hi Tim,

Happy New Year!

Thanks for the reminder. Have it updated.

Best,
Ying

On Thu, Jan 5, 2023 at 2:36 PM Tim Donohue ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/app/core/shared/media-viewer-item.model.ts
> <https://urldefense.com/v3/__https://github.com/DSpace/dspace-angular/pull/1967*discussion_r1062874687__;Iw!!BuQPrrmRaQ!jMEdV_Mx1jgC_5jLbdYI5wlx0CcwXjwpUjy3aBXO8eRxVW6hm6umRym-y-gLPAvVPyNE00aunilBuDMyz94IUgjpfA0cuw$>
> :
>
> > @@ -14,6 +14,11 @@ export class MediaViewerItem {
>     */
>    format: string;
>
> +  /**
> +   * Incoming Bitsream format mime type
> +   */
> +  minetype: string;
>
> @yingjin
> <https://urldefense.com/v3/__https://github.com/yingjin__;!!BuQPrrmRaQ!jMEdV_Mx1jgC_5jLbdYI5wlx0CcwXjwpUjy3aBXO8eRxVW6hm6umRym-y-gLPAvVPyNE00aunilBuDMyz94IUgjuE9z90Q$>
> : I expect this was lost/forgotten over the holidays (no worries, as I'm
> just returning as well). But I'd still recommend fixing the minor typo in
> this code (will also require fixing every location where this variable was
> used)
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/DSpace/dspace-angular/pull/1967*discussion_r1062874687__;Iw!!BuQPrrmRaQ!jMEdV_Mx1jgC_5jLbdYI5wlx0CcwXjwpUjy3aBXO8eRxVW6hm6umRym-y-gLPAvVPyNE00aunilBuDMyz94IUgjpfA0cuw$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AALAW7WGPF5XIOPCVBDLHKTWQ4WFFANCNFSM6AAAAAASCS3YH4__;!!BuQPrrmRaQ!jMEdV_Mx1jgC_5jLbdYI5wlx0CcwXjwpUjy3aBXO8eRxVW6hm6umRym-y-gLPAvVPyNE00aunilBuDMyz94IUgghkQ84Pg$>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

tdonohue reviewed (approved):
👍 Thanks @yingjin .  Tested this again today and it works as described.

Looking closer, I think the bug regarding having to enable `image` in order to get `video` to work is not going to be simple to fix. So, I think it needs to be treated as a separate bug ticket.  Based on how the `media.viewer.component.html` is written, it appears to assume that you can only enable `video` if `image` is also enabled.  I think it would require some refactoring to support something like:
```
mediaViewer:
  image: false
  video: true
```

So, I'll create a separate ticket for this bug, as it's not directly related to your work here.

tdonohue commented:
Merging with +1 approval for 7.5.  I've moved the outstanding bug to #2035 .  I've also added some basic documentation for this new captioning feature to https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Configuration#UserInterfaceConfiguration-MediaViewerSettings
